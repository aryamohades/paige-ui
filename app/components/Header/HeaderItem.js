import React from 'react';
import PropTypes from 'prop-types';

import { StyledLink, StyledButton, StyledButtonPrimary } from './itemStyles';

const HeaderItem = ({ primary, text, to, type }) => {
  let Wrapper = primary ? StyledButtonPrimary : StyledButton;

  if (type === 'link') {
    Wrapper = StyledLink;
  }

  return (
    <Wrapper to={to}>
      <span>{text}</span>
    </Wrapper>
  );
};

HeaderItem.propTypes = {
  primary: PropTypes.bool,
  text: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['link', 'button']),
};

HeaderItem.defaultProps = {
  type: 'link',
};

export default HeaderItem;
