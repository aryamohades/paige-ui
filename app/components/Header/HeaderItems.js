import React from 'react';
import PropTypes from 'prop-types';

import HeaderItem from './HeaderItem';

const HeaderItems = ({ items }) =>
  items.map(item => <HeaderItem key={item.id} {...item} />);

HeaderItems.propTypes = {
  items: PropTypes.array,
};

HeaderItems.defaultProps = {
  items: [],
};

export default HeaderItems;
