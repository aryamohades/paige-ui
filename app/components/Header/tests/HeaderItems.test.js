import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import HeaderItems from '../HeaderItems';
import HeaderItem from '../HeaderItem';

describe('<HeaderItems />', () => {
  it('should match the snapshot', () => {
    const items = [{ id: 1, to: '/', text: 'one' }];
    const renderedComponent = renderer
      .create(
        <Router>
          <HeaderItems items={items} />
        </Router>,
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should be empty if no items are passed', () => {
    const renderedComponent = mount(
      <Router>
        <HeaderItems />
      </Router>,
    );
    expect(renderedComponent.find(HeaderItem).length).toEqual(0);
  });

  it('should render 3 <HeaderItem /> components', () => {
    const items = [
      { id: 1, to: '/', text: 'one' },
      { id: 2, to: '/', text: 'two' },
      { id: 3, to: '/', text: 'three' },
    ];

    const renderedComponent = mount(
      <Router>
        <HeaderItems items={items} />
      </Router>,
    );
    expect(renderedComponent.find(HeaderItem)).toHaveLength(3);
  });

  it('should pass item props to rendered component', () => {
    const items = [
      { id: 1, to: '/', text: 'one' },
      { id: 2, to: '/', text: 'two' },
    ];

    const renderedComponent = mount(
      <Router>
        <HeaderItems items={items} />
      </Router>,
    );

    expect(
      renderedComponent
        .find(HeaderItem)
        .at(0)
        .prop('text'),
    ).toBe(items[0].text);
    expect(
      renderedComponent
        .find(HeaderItem)
        .at(1)
        .prop('text'),
    ).toBe(items[1].text);
  });
});
