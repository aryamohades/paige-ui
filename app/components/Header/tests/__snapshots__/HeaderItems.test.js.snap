// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderItems /> should match the snapshot 1`] = `
<a
  className="itemStyles__StyledLink-fAIfBS gvqALX"
  href="/"
  onClick={[Function]}
>
  <span>
    one
  </span>
</a>
`;
